r2i/onnxrt_openvino/parameters.h

```diff
@@ -0,0 +1,161 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#ifndef R2I_ONNXRT_OPENVINO_PARAMETERS_H
+#define R2I_ONNXRT_OPENVINO_PARAMETERS_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include <r2i/iparameters.h>
+#include <r2i/runtimeerror.h>
+#include <r2i/onnxrt_openvino/engine.h>
+#include <r2i/onnxrt/parameters.h>
+
+namespace r2i {
+namespace onnxrt_openvino {
+
+class Parameters: public r2i::onnxrt::Parameters {
+ public:
+  Parameters ();
+  RuntimeError Configure (std::shared_ptr<IEngine> in_engine,
+                          std::shared_ptr<IModel> in_model) override;
+  std::shared_ptr<IEngine> GetEngine () override;
+  RuntimeError Get (const std::string &in_parameter, int &value) override;
+  RuntimeError Get (const std::string &in_parameter, double &value) override;
+  RuntimeError Get (const std::string &in_parameter,
+                    std::string &value) override;
+  RuntimeError Set (const std::string &in_parameter,
+                    const std::string &in_value) override;
+  RuntimeError Set (const std::string &in_parameter, int in_value) override;
+  RuntimeError Set (const std::string &in_parameter, double in_value) override;
+  RuntimeError List (std::vector<ParameterMeta> &metas) override;
+
+ private:
+  std::shared_ptr <Engine> engine;
+
+  friend class Accessor;
+
+  class Accessor {
+   public:
+    Accessor (Parameters *target) : target(target) {}
+    virtual RuntimeError Set () = 0;
+    virtual RuntimeError Get () = 0;
+    virtual ~Accessor() {}
+
+   protected:
+    Parameters *target;
+  };
+
+  class StringAccessor : public Accessor {
```

Comment:
Why are all this here and we are not making use of the ones in the parent?

Reply:
Answering this one and all related comments above. This is because you [can not inherite friendship in C++](https://stackoverflow.com/questions/3561648/why-does-c-not-allow-inherited-friendship#:~:text=A%20derived%20class%20can%20inherit,'member'%20of%20the%20base.&text=%2411.4%20%2D%20%22Also%2C%20because%20the,from%20the%20class%20granting%20friendship.%22), so basically nothing that uses the Accesor class in derived classes.